app/src/main/java/org/sil/gatherwords/SessionActivity.java

```diff
@@ -1,127 +1,134 @@
-
-        Intent i = new Intent(this, MainActivity.class);
-        startActivity(i);
-    }
-
-    public void input_location(View view) {
-        if (locationEnabled) {
-            locationEnabled = false;
-            location = null;
-        } else {
-            mFusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
-            location = new Location("vanDellen 362");
-            locationEnabled = false;
-
-            if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-                ActivityCompat.requestPermissions(this,
-                        new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
-                        1);
-            } else {
-                locationEnabled = true;
-            }
-
-            if (locationEnabled) {
-                setSessionLocation();
-            }
-        }
-    }
-
-    private void setSessionLocation() {
-        LocationManager mLocationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
-
-        if (!mLocationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
-            SwitchCompat sw = findViewById(R.id.session_create_location_swtich);
-            Snackbar mySnackbar = Snackbar.make(findViewById(R.id.session_create_layout),
-                    "Please enable location services to access this feature", Snackbar.LENGTH_LONG);
-            mySnackbar.show();
-            sw.setChecked(false);
-            locationEnabled = false;
-        } else {
-            location = mLocationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
-            //TODO: Something with this location
+	private FusedLocationProviderClient mFusedLocationClient;
+	boolean locationEnabled;
+	Location location;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_session);
+
+		locationEnabled = false;
+
+		long date = System.currentTimeMillis() / 1000L;
+
+		EditText dateField = findViewById(R.id.session_create_date);
+		dateField.setText(Long.toString(date));
+		dateField.setEnabled(false);
+
+
+	}
+
+	public void save_settings_fab_pressed(View view) {
+		EditText name = findViewById(R.id.session_create_name);
+		EditText eliciter = findViewById(R.id.session_create_eliciter);
+		EditText speaker = findViewById(R.id.session_create_speaker);
+		EditText date = findViewById(R.id.session_create_date);
+
+		Session session = new Session();
+		session.label = name.getText().toString();
+		session.recorder = eliciter.getText().toString();
+		session.speaker = speaker.getText().toString();
+		// TODO: decide on internal format
+		// session.date = date.getText().toString();
+
+		// Create db instance and insert the session
+		AppDatabase appDatabase = Room.databaseBuilder(getApplicationContext(), AppDatabase.class, getString(R.string.database_name)).build();
```

Comment:
Okay for the time being, I suppose.  However, we will want to switch db access to a singleton, as `.build()` is very expensive.